File: Napakalaki-Game(PDOO)/NapakalakiRuby/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'monster.rb'
class CultistPlayer < Player
  
  @@totalCultistPlayers=0
  
  def initialize(p,c)
    super(p)
       @myCultistCard=c
       @@totalCultistPlayers=@@totalCultistPlayers+1
  end
  
    def getCombatLevel()
       salida=((super+super*0.2 + @myCultistCard.getGainedLevels())*@@totalCultistPlayers).to_i
       return salida
    end
    
    def getOponentLevel(m)
        salida=m.getCombatLevelChangeAgainstCultistPlayer()
    return salida
    end
    
    def shouldConvert()
        salida=false
        return salida
    end
 
    def giveMeATreasure()
        r = Random.new
        numero = r.rand(0..@visibleTreasures.size())
        tesoro= @visibleTreasures.at(numero)
       return tesoro
    end
    
    def canYouGiveMeATreasure()
       salida=true
      if(@visibleTreasures.empty? && @hiddenTreasures.empty?)
          salida=false
      end
     return salida
    end
    
    public
    def getTotalCultistPlayers()
      return @totalCultistPlayers;
    end
end

File: Napakalaki-Game(PDOO)/NapakalakiJava/src/napakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalakiGame;

/**
 *
 * @author joaquin
 */
public class CultistPlayer extends Player {
    private Cultist myCultistCard;
    private static int totalCultistPlayers=0;
    
    public CultistPlayer(Player p ,Cultist c){
       super(p);
       this.myCultistCard=c;
       totalCultistPlayers++;
    }
    
    @Override
    public int getCombatLevel(){
    double salida=0;
    salida=(super.getCombatLevel()+super.getCombatLevel()*0.2 + myCultistCard.getGainedLevels())*totalCultistPlayers;
    return (int) salida;
    }
    @Override
    protected int getOponentLevel(Monster m){
        int salida;
        salida=m.getCombatLevelAgainstCultistPlayer();
    return salida;
    }
    @Override
    protected boolean shouldConvert(){
    return false;
    }
    @Override
    protected Treasure giveMeATreasure(){
           Treasure tesoro;
      int aleatorio = visibleTreasures.size();
      int numero= (generator.nextInt(aleatorio));
                 tesoro=visibleTreasures.get(numero);
  return tesoro;
    
    }
    @Override
    protected boolean canYouGiveMeATreasure(){
    boolean salida=true;
      if(visibleTreasures.isEmpty()&&hiddenTreasures.isEmpty())
          salida=false;
  return salida;
    
    }
    @Override
    public boolean getConvertido(){
  return true;
  }
    @Override
    public int getTotalCultistPlayers(){
    return totalCultistPlayers;
    }
    
}
